Document the GitHub (read-only) integration (#773)
Promptless/docs: src/content/docs/docs/connect/source-control.mdx

Only GitHub-based collections receive documentation pull requests. GitLab and Bitbucket connect as trigger sources (and, for GitLab, context sources): Promptless monitors their pull and merge requests but does not publish documentation to them.

> GitHub also offers a read-only install mode — the [GitHub (read-only) app](/docs/reference/integrations/github-read-only) — that grants read access and webhooks only and publishes to public repositories through a fork.

Reviewer: ```suggestion
Promptless also has a [read-only GitHub app](/docs/reference/integrations/github-read-only) that grants read access and webhooks only and publishes to public repositories through a fork.
```
